src/swdmt/redistricting/StarGeneratorTest.java

```diff
@@ -0,0 +1,36 @@
+package swdmt.redistricting;
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+class StarGeneratorTest {
+    /**
+     * Ensure each pattern generated has the correct number of Locations,
+     * and there are no duplicates in any of the generated patterns.
+     */
+    @Test
+    void generatePatternTest() {
+        int n = 100;
+        ArrayList<ArrayList<Location>> patterns = StarGenerator.generatePattern(n);
+        for(ArrayList<Location> pattern: patterns) {
+            assertEquals(pattern.size(), n);
+            Set<Location> patternSet = new HashSet<>();
+            Location previousDistrict = null;
+            for(Location currentDistrict: pattern) {
+                if(previousDistrict != null) {
+                    assertTrue(
+                            currentDistrict.xCoordinate() + 1 == previousDistrict.xCoordinate() ||
+                                    currentDistrict.xCoordinate() - 1 == previousDistrict.xCoordinate() ||
+                                    currentDistrict.yCoordinate() + 1 == previousDistrict.yCoordinate() ||
+                                    currentDistrict.yCoordinate() - 1 == previousDistrict.yCoordinate()
+                    );
+                }
+                previousDistrict = currentDistrict;
```

Comment:
Why is an object of type `Location` referred to as a _district_ (`previousDistrict`,`currentDistrict`)?
This code is not intuitive so needs explanation or rewriting.
Is the intent to show that no two districts are identical? (Does this suggest we need an `equals` predicate defined for the `District` class?)
Is the intent to show that no district contains two identical locations? (Could we use some existing `contains` method or some Set operation?)